server.js

```diff
@@ -9,12 +12,113 @@ app.use(cors());
 //Use this array as your (in-memory) data store.
 const bookings = require("./bookings.json");
 
-app.get("/", function (request, response) {
+app.get("/",  (request, response) => {
   response.send("Hotel booking server.  Ask for /bookings, etc.");
 });
 
-// TODO add your routes and helper functions here
 
-const listener = app.listen(process.env.PORT, function () {
-  console.log("Your app is listening on port " + listener.address().port);
+// get all bookings
+
+  app.get("/bookings", (req, res) => {
+    res.status(200);
+    res.send(bookings);
+  });
+
+// Get one booking by id
+app.get("/bookings/:id", (req, res) => {
+  const id = Number(req.params.id);
+  const foundBooking = bookings.find((booking) => {
+    return booking.id === id;
+    // returns element || undefined
+  });
+
+  foundBooking !== undefined
+    ? res.status(200).send(foundBooking)
+    : res.status(404).send(`There is no booking with id ${id}`);
+});
+
+
+// Create a new booking
+  app.post("/bookings", (req, res) => {
+    const schema = Joi.object({
+      Title: Joi.string().min(2).required(),
+      FirstName: Joi.string().max(50).required(),
+      Surname: Joi.string().max(50).required(),
+      Email: Joi.string().email().required(),
+      RoomId: Joi.number().integer().min(1).required(),
+      CheckInDate: Joi.date().iso().required(),
+      CheckOutDate: Joi.date().iso().greater(Joi.ref("CheckInDate")).required(),
+    });
+
+    const result = schema.validate(req.body);
+
+    const {
+      Title,
+      FirstName,
+      Surname,
+      Email,
+      RoomId,
+      CheckInDate,
+      CheckOutDate,
+    } = req.body;
+
+    const generateRandomId = () => Date.now(); // id for the new booking
```

Comment:
I like your solution of using a timestamp for a unique ID. There are also libraries that you can use that will generate unique IDs